scripts/deps.py

```diff
@@ -1,24 +1,32 @@
-import sys
+import jinja2
+import yaml
 
-from conda_build import api
 
-section = {
-    'build': 'requirements/build',
-    'run':   'requirements/run',
-    'test':  'test/requires',
-}
+def load_setup_py_data():
+    import os
+    import setuptools
+    os.environ['CONDA_BUILD_STATE'] = 'RENDER'
+    data = {}
 
-names = set(sys.argv[1:])
+    def _setup(**kw): data.update(kw)
+    setuptools.setup = _setup
+    return data
 
-if not all(name in section for name in names):
-    print("Got unknown section name. Valid section names are: build, run, or test")
-    sys.exit(1)
-
-metadata, _, _ = api.render('conda.recipe')
+meta = jinja2.Template(open("conda.recipe/meta.yaml").read())
+out = meta.render(load_setup_py_data=load_setup_py_data)
+yam = yaml.load(out)
 
 deps = ""
+_list = yam["requirements"]["build"]
+for item in yam["test"]["requires"] + yam["requirements"]["run"]:
+    if item not in _list:
+        _list.append(item)
+
+deps += " ".join(s for s in _list)
+deps = deps.replace(',', '')
```

Comment:
is this replacement actually needed? Do individual items in `_list` have ever commas inside them? (not rhetorical, I don't know, but I guess I'd be surprised if they did)